src/installer/tests/HostActivation.Tests/PortableAppActivation.cs

```diff
@@ -193,21 +193,20 @@ public void AppHost_FrameworkDependent_GlobalLocation_Succeeds(bool useRegistere
 
                 // Verify running with the default working directory
                 Command.Create(appExe)
-                    .CaptureStdErr()
-                    .CaptureStdOut()
+                    .EnableTracingAndCaptureOutputs()
                     .MultilevelLookup(false)
                     .ApplyRegisteredInstallLocationOverride(registeredInstallLocationOverride)
                     .EnvironmentVariable(Constants.TestOnlyEnvironmentVariables.DefaultInstallPath, useRegisteredLocation ? null : builtDotnet)
                     .DotNetRoot(null)
                     .Execute()
                     .Should().Pass()
                     .And.HaveStdOutContaining("Hello World")
-                    .And.HaveStdOutContaining(sharedTestState.RepoDirectories.MicrosoftNETCoreAppVersion);
+                    .And.HaveStdOutContaining(sharedTestState.RepoDirectories.MicrosoftNETCoreAppVersion)
+                    .And.NotHaveStdErrContaining("Failed to read environment variable");
```

Comment:
Can you try switching these to just check `NotHaveStdErr()`? The general issue was that we were writing to stderr at all on success. I believe that the environment variable message was the only thing we are incorrectly writing out, so we shouldn't have anything in stderr with this fix.

Reply:
StdErr will not be empty  : there's an other warning in StdErr that is not related to reading env variable